patternfly/patternfly-elements
# feat: pfe-tabs horizontal overflow scrolling

Closes #1013 

## Description
Adds horizontal scrolling capabilities to horizontal tabs. This feature adds a left arrow and a right arrow when the tabs are too large for its container. This feature uses a ResizeObserver to respond to changes in the size of the component and will add the scrolling capability once the tabs are too large. This change also impacts `pfe-content-set` in that when the component renders tabs, they will have the scrolling feature if the tabs are too large.

### Related issues
#1013 


### Preview
- [Horizontal scrolling demo](https://deploy-preview-1698--patternfly-elements.netlify.app/elements/pfe-tabs/demo/horizontal-scroll.html)
- [Tabs demo](https://deploy-preview-1698--patternfly-elements.netlify.app/components/tabs/)
- [Tabs in pfe-content-set](https://deploy-preview-1698--patternfly-elements.netlify.app/elements/pfe-content-set/demo/) Scroll to the "The pfe-content-set comes with a mutation observer so that you may dynamically add headers and panels." section and continue to click the "Add Header & Panel" button until the tabs are too big for the container

### Testing instructions

- [ ] *pfe-tabs*
  1. Go to https://deploy-preview-1698--patternfly-elements.netlify.app/elements/pfe-tabs/demo/horizontal-scroll.html
  2. The first tab set should have a horizontal scroll bar
  3. Scroll all the way to the right
  4. Make sure that the right arrow on the side of the tabs is disabled and the left arrow is enabled
  5. Scroll all the way to the left
  6. Make sure the left arrow is disabled and the right arrow is enabled
  7. Scroll to the middle
  8. Make sure both the left arrow and the right arrow are enabled
  9. Scroll all the way to the left
  10. Click the right arrow
  11. Make sure that the previously hidden content is now visible
  12. Click the right arrow until it scrolls all the way to the right.
  13. Click the left arrow 
  14. Make sure the previously hidden content is now visible
  15. Click the left arrow until is scrolls all the way to the left
  16. Resize the screen and make sure that the left and right arrows are added to tab sets that are too large for the container
  17. Make the screen larger and make sure the arrows are now longer visible when the tab set is small enough to fit inside its container
- [ ] *pfe-content-set*
  1. Go to https://deploy-preview-1698--patternfly-elements.netlify.app/elements/pfe-content-set/demo/
  2. Scroll to the "The pfe-content-set comes with a mutation observer so that you may dynamically add headers and panels." section and continue to click the "Add Header & Panel" button until the tabs are too big for the container
  3. Make sure the left and right arrows appear when the tabs are too large for its container


#### Browser requirements

Your component should work in all of the following environments:

- [ ] Internet Explorer 11 (should be useable, not pixel perfect. There should be a horiztonal scroll bar when the tabs are too large for the container)


### Ready-for-merge Checklist





### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
